docs/userdoc/Iterators.md

```diff
@@ -1,49 +1,169 @@
+  If the internal data structure is updated or changed, the iteration order can change as well.
+- Alternatively to `iter != autoPas.end()`, we can also use `iter.isValid()`.
+  This is equivalent, because `AutoPas::end()` returns a `bool` and `ContainerIterator::operator==(bool)` calls `ContainerIterator::isValid()`.
+
+#### Particle Deletion
+
+AutoPas supports the deletion of particles while iterating without invalidating the iterator.
+However, this needs to be done through the member function `AutoPas::deleteParticle()`.
+This function also leaves the iterator in a state so that `ContainerIterator::operator++()` needs to be called.
+
+```cpp
+for(auto iter = autoPas.begin(); iter != autoPas.end(); ++iter) {
+  autoPas.deleteParticle(iterator);
+}
+```
+
+### Restricting Particle Types
+
+Iterators can be restricted to only return particles of a given ownership type.
+The default is to cover owned and halo particles.
+See [ParticleOwnershipModel.md](https://github.com/AutoPas/AutoPas/blob/master/docs/userdoc/ParticleOwnershipModel.md) for particle ownership types.
+
+```cpp
+// only iterate owned particles
+for(auto iter = autoPas.begin(autopas::IteratorBehavior::owned);
+    iter != autoPas.end();
+    ++iter) {
+  // user code
+}
+```
+
+Depending on the container, this might be more efficient than iterating everything and calling `continue` on undesired ownership types.
+
+### Region Iterators
+
+Iterators can be restricted to only return particles inside of a given region.
+Regions are always cuboids defined by their left-lower-frontal and right-upper-back corner.
+
+```cpp
+// only iterate particles in a given box
+const std::array<double, 3> lowCorner = ...;
+const std::array<double, 3> highCorner = ...;
+for(auto iter = autoPas.getRegionIterator(lowCorner, highCorner);
+    iter != autoPas.end();
+    ++iter) {
+  // user code
+}
+```
+
+Region Iterators also take an `IteratorBehavior` as an optional third argument.
+
+### Parallelization
+
+Iteration of the AutoPas object can also be done in parallel via OpenMP.
+For this, place any of the examples above in a parallel region.
 
-To iterate over a subset of particles, the `getRegionIterator(lowCorner, highCorner)` method can be used:
 ```cpp
 AUTOPAS_OPENMP(parallel)
-for(auto iter = autoPas.getRegionIterator(lowCorner, highCorner); iter != autoPas.end(); ++iter) {
-  // user code:
-  auto position = iter->getR();
+for(auto& particle : autoPas) {
+  // user code
 }
 ```
 
-Both `begin()` and `getRegionIterator()` can also take the additional parameter `IteratorBehavior`, which indicates over which particles the iteration should be performed. See [autopas::IteratorBehavior](https://autopas.github.io/doxygen_documentation/git-master/classautopas_1_1options_1_1IteratorBehavior.html) for possible options and details.
-The default parameter is `ownedOrHalo`, which is also used for range-based for loops.
+Note that this is equal to `#pragma omp parallel` and **NOT** a `parallel for` clause!
```

Comment:
I don't really care if users use them. They are just a convenience if you want to avoid compiler warnings about unknown macros in some cases. Hence, I wouldn't mention them here since they then need some explanation.

Reply:
Okay, then it is not so important.